### PR TITLE
(maint) quote file path in `file_contents_on` and `file_exists_on`

### PR DESCRIPTION
**Pull request by ciprianbadescu**

### CODE REVIEW

File: lib/beaker/dsl/helpers/host_helpers.rb [hunk 1]
```diff
@@ -497,7 +497,7 @@ def file_exists_on(host, file_path)
             if file_path.include?(':')
               split_path = win_ads_path(file_path)
 
-              command = %(Test-Path #{split_path[:path]})
+              command = %(Test-Path "#{split_path[:path]}")
```
Comment (sbeaulie) [inline lib/beaker/dsl/helpers/host_helpers.rb:500]: Is this to support paths with spaces in them? 
I think you would need it here too and on line 495 above for the case where there is no ads path?

https://github.com/puppetlabs/beaker/pull/1647/files#diff-cdd2e965e7b543eb19634118d43851b8R495

File: lib/beaker/dsl/helpers/host_helpers.rb [hunk 2]
```diff
@@ -555,7 +555,7 @@ def file_contents_on(host, file_path)
             if host['platform'] =~ /windows/
               file_path.gsub!('/', '\\')
```
Comment (sbeaulie) [inline lib/beaker/dsl/helpers/host_helpers.rb:556]: that substitution is not done in file_exists_on

File: lib/beaker/dsl/helpers/host_helpers.rb [hunk 3]
```diff
@@ -555,7 +555,7 @@ def file_contents_on(host, file_path)
...
 
-              command = %{Get-Content -Raw -Path #{file_path}}
+              command = %{Get-Content -Raw -Path "#{file_path}"}
               command += %{ -Stream #{split_path[:ads]}} if split_path[:ads]
```
Comment (sbeaulie) [inline lib/beaker/dsl/helpers/host_helpers.rb:559]: would you need it here too?

**Comment by ciprianbadescu** [comment]
To give more background, puppet-code acceptance failed with bellow error. It looks that if the "c:\\" path is passed as parameter from directly from cygwin to powershell without " it gets messed up.

I updated in this PR the two cases hit by my tests (while I still think `file_exists_on` is not really working, as it never fails, with or without the change)
```
    naval-racetrack.delivery.puppetlabs.net (windows2012r2-64-1) 15:38:23$ powershell.exe -ExecutionPolicy Bypass -InputFormat None -NoLogo -NoProfile -NonInteractive -Command Test-Path C:\Windows\TEMP\install-puppet-2020-05-15_15.38.14.log
      False
    
    naval-racetrack.delivery.puppetlabs.net (windows2012r2-64-1) executed in 0.92 seconds
    
    naval-racetrack.delivery.puppetlabs.net (windows2012r2-64-1) 15:38:24$ powershell.exe -ExecutionPolicy Bypass -InputFormat None -NoLogo -NoProfile -NonInteractive -Command Get-Content -Raw -Path C:\Windows\TEMP\install-puppet-2020-05-15_15.38.14.log
      Get-Content : Cannot find path 'C:\cygwin64\home\Administrator\WindowsTEMPinsta
      ll-puppet-2020-05-15_15.38.14.log' because it does not exist.
      At line:1 char:1
      + Get-Content -Raw -Path C:WindowsTEMPinstall-puppet-2020-05-15_15.38.14.log
      + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          + CategoryInfo          : ObjectNotFound: (C:\cygwin64\hom...15_15.38.14.l 
         og:String) [Get-Content], ItemNotFoundException
          + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetCo 
         ntentCommand
       
```

File: lib/beaker/dsl/helpers/host_helpers.rb (hunk 3 above)
Comment (ciprianbadescu) [reply lib/beaker/dsl/helpers/host_helpers.rb:559]: Yes, this change is surely needed

File: lib/beaker/dsl/helpers/host_helpers.rb (hunk 1 above)
Comment (ciprianbadescu) [reply lib/beaker/dsl/helpers/host_helpers.rb:500]: It is to protect windows path (c:\asdasd) from cygwin shell expansion. Without it `C:\Windows\TEMP\install-puppet-2020-05-15_15.38.14.log` gets modified to `C:\cygwin64\home\Administrator\WindowsTEMPinstall-puppet-2020-05-15_15.38.14.log`

File: lib/beaker/dsl/helpers/host_helpers.rb (hunk 2 above)
Comment (ciprianbadescu) [reply lib/beaker/dsl/helpers/host_helpers.rb:556]: It looks to work without, should I still add it there?

**Comment by donoghuc** [comment]
This looks very similar to https://github.com/puppetlabs/beaker/pull/1626#issuecomment-590594406 are you using a beaker-puppet with this change https://github.com/puppetlabs/beaker-puppet/pull/135

**Comment by ciprianbadescu** [comment]
@donoghuc , I will recheck as the PR you mention was not merged/released at that time.

**Comment by ciprianbadescu** [comment]
Problem is already solved in beaker 4.23.1

**Pull request closed without merging**